### ISSUE TITLE
Unjustified warning for D.3

### ISSUE THREAD

**Issue by llvmbot** [llvm/llvm-project#23446]
|  |  |
| --- | --- |
| Bugzilla Link | [23072](https://llvm.org/bz23072) |
| Version | trunk |
| OS | Linux |
| Reporter | LLVM Bugzilla Contributor |
| CC | @dwblaikie,@zygoloid |

## Extended Description 
The problem is that an unjustified warning is arisen when explicitly defaulted copy-constructor is defined when the class has a user-declared copy
assignment operator or a user-declared destructor. $D.3 states it's about implicit and not explicit definition. Quote:

The implicit definition of a copy constructor as defaulted is deprecated if the class has a user-declared copy
assignment operator or a user-declared destructor. 

Code like this:
```cpp
struct W {

  W(const W&) = default;  

  int a;

  ~W() { a = 9; }
};
```
will emit unjustified warning for now.

**Comment by AaronBallman** [comment]
I am not able to reproduce the false positive warnings, so closing this as works for me. However, if the issue still reproduces for you, please reopen with more details on how to reproduce it.
